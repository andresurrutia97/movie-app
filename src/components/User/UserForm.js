import React from 'react';
import PropTypes from 'prop-types';

import Input from '../Ui/InputContainer';

const UserForm = ({ elements, onChange }) => {
  const form = elements.map((el) => {
    return (
      <Input
        key={el.id}
        title={el.config.elementName}
        elementtype={el.config.elementType}
        elementConfig={el.config.elementConfig}
        value={el.config.value}
        changed={(event) => onChange(event, el.id)}
        shouldValidate={el.config.validation}
        invalid={!el.config.valid}
        touched={el.config.touched}
      />
    );
  });
  return <div className="w-full"> {form}</div>;
};

UserForm.propTypes = {
  elements: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      title: PropTypes.string,
      elementtype: PropTypes.string,
      elementConfig: PropTypes.array,
      value: PropTypes.string,
      changed: PropTypes.func,
      shouldValidate: PropTypes.bool,
      invalid: PropTypes.bool,
      touched: PropTypes.bool,
    })
  ),
  onChange: PropTypes.func,
};

UserForm.defaultProps = {
  elements: [],
  onChange: () => {},
};

export default UserForm;
