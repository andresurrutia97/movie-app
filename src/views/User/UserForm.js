export default {
  name: {
    elementType: 'input',
    elementName: 'Name',
    elementConfig: {
      type: 'text',
      placeholder: 'Name',
    },
    value: '',
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
    fullWidth: false,
  },
  tel: {
    elementType: 'input',
    elementName: 'Phone',
    elementConfig: {
      type: 'number',
      placeholder: 'Phone number',
    },
    value: '',
    validation: {
      required: true,
      isNumeric: true,
    },
    valid: false,
    touched: false,
    fullWidth: false,
  },
  img: {
    elementType: 'input',
    elementName: 'Picture',
    elementConfig: {
      type: 'text',
      placeholder: 'Picture url',
    },
    value: '',
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
    fullWidth: false,
  },
  desc: {
    elementType: 'textarea',
    elementName: 'Description',
    elementConfig: {
      type: 'text',
      placeholder: 'Description',
    },
    value: '',
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
    fullWidth: true,
  },
};
