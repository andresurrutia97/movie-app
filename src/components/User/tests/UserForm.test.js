import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import UserForm from '../UserForm';
import Input from '../../Ui/InputContainer';
import form from '../../../views/User/UserForm';

describe('<UserForm/>', () => {
  const elementsMock = Object.keys(form).map((control) => {
    return { id: control, config: form[control] };
  });

  const component = <UserForm elements={elementsMock} />;

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain 4 instances of <Input/>', () => {
    const wrapper = mount(component);
    expect(wrapper.find(Input)).toHaveLength(4);
  });
});
