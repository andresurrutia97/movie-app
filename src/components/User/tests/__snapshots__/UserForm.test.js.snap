// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm/> should render correctly  1`] = `
<div
  className="w-full"
>
   
  <div
    className="mb-1"
  >
    <div
      className="text-white mb-2"
    >
      Name
    </div>
    <input
      className="box-border bg-black appearance-none rounded border border-2 custom-blue-border w-full py-3 px-4 text-white mb-3 leading-tight focus:outline-none"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
  <div
    className="mb-1"
  >
    <div
      className="text-white mb-2"
    >
      Phone
    </div>
    <input
      className="box-border bg-black appearance-none rounded border border-2 custom-blue-border w-full py-3 px-4 text-white mb-3 leading-tight focus:outline-none"
      onChange={[Function]}
      placeholder="Phone number"
      type="number"
      value=""
    />
  </div>
  <div
    className="mb-1"
  >
    <div
      className="text-white mb-2"
    >
      Picture
    </div>
    <input
      className="box-border bg-black appearance-none rounded border border-2 custom-blue-border w-full py-3 px-4 text-white mb-3 leading-tight focus:outline-none"
      onChange={[Function]}
      placeholder="Picture url"
      type="text"
      value=""
    />
  </div>
  <div
    className="mb-1"
  >
    <div
      className="text-white mb-2"
    >
      Description
    </div>
    <input
      className="box-border bg-black appearance-none rounded border border-2 custom-blue-border w-full py-3 px-4 text-white mb-3 leading-tight focus:outline-none"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
  </div>
</div>
`;
